randovania/game_description/resources/resource_info.py

```diff
@@ -28,7 +29,7 @@ def add_resource_gain_to_current_resources(resource_gain: ResourceGain,
 
 
 def add_resources_into_another(target: CurrentResources, source: CurrentResources) -> None:
-    resource_gain: ResourceGain = source.items()
+    resource_gain = typing.cast(ResourceGain, source.items())
```

Comment:
what's the difference here?

Reply:
The first I'm declaring the variable is of type `ResourceGain`, but I can still get a warning if the argument isn't.
The new one I'm explicitly casting the variable to the new type. Kinda like asserting, but it could be wrong.

From what I understand, `source.items()` really should match the `ResourceGain` type, but at least PyCharm gets lost somewhere in the types and gives me a warning.